### PR TITLE
feat(connector): [Mollie] Currency Unit Conversion

### PR DESCRIPTION
**Pull request by SagarDevAchar**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Addressing Issue: #2231 
- Modified two files in `hyperswitch/crates/router/src/connector/`
  - `mollie.rs`
    - Implement `get_currency_unit` function
    - Modify `ConnectorIntegration` implementations for `Mollie`
  - `mollie/transformers.rs`
    - Implement `MollieRouterData<T>` structure and functionality
    - Modify implementation for `MolliePaymentsRequest`

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

We need to create a payment using mollie and test the currency unit from connector response and connector dashboard.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/mollie/transformers.rs [hunk 1]
```diff
@@ -120,50 +152,58 @@ pub struct MollieBrowserInfo {
...
+        item: &MollieRouterData<&types::PaymentsAuthorizeRouterData>,
+    ) -> Result<Self, Self::Error> {
         let amount = Amount {
-            currency: item.request.currency,
-            value: utils::to_currency_base_unit(item.request.amount, item.request.currency)?,
+            currency: item.router_data.request.currency,
+            value: utils::to_currency_base_unit(
+                item.router_data.request.amount,
+                item.router_data.request.currency,
+            )?,
```
Comment (SamraatBansal) [inline crates/router/src/connector/mollie/transformers.rs:165]: You need not to call this function `utils::to_currency_base_unit` as we have already converted the `amount`. You can directly use that.

```suggestion
            value: item.amount
```

File: crates/router/src/connector/mollie/transformers.rs [hunk 2]
```diff
@@ -526,16 +566,21 @@ pub struct MollieRefundRequest {
...
+        item: &MollieRouterData<&types::RefundsRouterData<F>>,
+    ) -> Result<Self, Self::Error> {
         let amount = Amount {
-            currency: item.request.currency,
-            value: utils::to_currency_base_unit(item.request.refund_amount, item.request.currency)?,
+            currency: item.router_data.request.currency,
+            value: utils::to_currency_base_unit(
+                item.router_data.request.refund_amount,
+                item.router_data.request.currency,
+            )?,
```
Comment (SamraatBansal) [inline crates/router/src/connector/mollie/transformers.rs:579]: You need not to call this function `utils::to_currency_base_unit` as we have already converted the `amount`. You can directly use that.

```suggestion
            currency: item.router_data.request.currency,
            value: item.amount,
```

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/mollie/transformers.rs (hunk 1 above)
Comment (SagarDevAchar) [reply crates/router/src/connector/mollie/transformers.rs]: Updated!

File: crates/router/src/connector/mollie/transformers.rs (hunk 2 above)
Comment (SagarDevAchar) [reply crates/router/src/connector/mollie/transformers.rs]: Updated!

**Review by SamraatBansal — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Comment by swangi-kumari** [comment]
Hey @SagarDevAchar ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**